### PR TITLE
Upgrade react query

### PR DESCRIPTION
**Pull request by dtrucs**
- Bump @tanstack/react-query from 4.10.3 to 5.51.1
- Call `commonDictionaries` with fetchQuery

### CODE REVIEW

File: frontend/src/pages/outdoor-site/[outdoorSite].tsx [hunk 1]
```diff
@@ -14,14 +15,27 @@ export const getServerSideProps: GetServerSideProps = async context => {
...
+      queryKey: ['commonDictionaries', locale],
+      queryFn: () => getCommonDictionaries(locale),
+    });
 
     const details = await getOutdoorSiteDetails(id, locale, commonDictionaries);
-    await queryClient.prefetchQuery(['outdoorSiteDetails', id, locale], () => details);
+    await queryClient.prefetchQuery({
+      queryKey: ['outdoorSiteDetails', id, locale],
+      queryFn: () => details,
+    });
```
Comment (mabhub) [inline frontend/src/pages/outdoor-site/[outdoorSite].tsx:38]: Maybe:

```suggestion
    await queryClient.prefetchQuery({
      queryKey: ['outdoorSiteDetails', id, locale],
      queryFn: () => getOutdoorSiteDetails(id, locale, commonDictionaries),
    });
    
    const details = queryClient.getQueryData(['outdoorSiteDetails', id, locale]);
```

**Review by mabhub — APPROVED** [review]

File: frontend/src/components/ActivitySearchFilter/useActivitySearchFilter.ts [hunk 2]
```diff
@@ -10,9 +10,11 @@ const { activityBar } = getHomePageConfig();
 
 export const useActivitySearchFilter = () => {
   const language = useRouter().locale ?? getDefaultLanguage();
-  const { data: activities } = useQuery<ActivityFilter[], Error>(['homeActivities', language], () =>
-    getActivityBarContent(language, activityBar.links),
-  );
+  const { data: activities } = useQuery<ActivityFilter[], Error>({
+    queryKey: ['homeActivities', language],
+
```
Comment (mabhub) [inline frontend/src/components/ActivitySearchFilter/useActivitySearchFilter.ts:15]:
```suggestion
```

File: frontend/src/components/pages/details/useDetails.tsx [hunk 3]
```diff
@@ -43,39 +40,29 @@ export const useDetails = (
...
-    ['trekFamily', parentIdString, language],
-    () => getTrekFamily(isUrlString(parentId) ? parentId : '', language),
-    {
-      enabled: isUrlString(parentId),
-      staleTime: ONE_DAY,
-    },
-  );
+  const { data: trekFamily } = useQuery<TrekFamily | null, Error>({
+    queryKey: ['trekFamily', parentIdString, language],
+    queryFn: () => getTrekFamily(isUrlString(parentId) ? parentId : '', language),
```
Comment (mabhub) [inline frontend/src/components/pages/details/useDetails.tsx:62]:
```suggestion
    queryFn: () => getTrekFamily(parentIdString, language),
```

File: frontend/src/components/pages/flatPage/useFlatPage.tsx [hunk 4]
```diff
@@ -22,25 +22,23 @@ export const useFlatPage = (flatPageUrl: string | undefined) => {
...
     'ids' in suggestion ? suggestion.ids : [suggestion.type],
   );
 
-  const { data: activitySuggestions = [] } = useQuery<ActivitySuggestion[] | [], Error>(
-    ['activitySuggestions', ...activitySuggestionIds, id, language],
-    () => getActivitySuggestions(suggestions, language, commonDictionaries),
-    { enabled: suggestions.length > 0 && commonDictionaries !== undefined },
-  );
+  const { data: activitySuggestions = [] } = useQuery<ActivitySuggestion[] | [], Error>({
+    queryKey: ['activitySuggestions', ...activitySuggestionIds, id, language],
```
Comment (mabhub) [inline frontend/src/components/pages/flatPage/useFlatPage.tsx:38]:
```suggestion
    queryKey: ['activitySuggestions', activitySuggestionIds, id, language],
```

No need to spread `activitySuggestionIds` as TanStack Query serialize it: https://tanstack.com/query/latest/docs/framework/react/guides/query-keys

File: frontend/src/components/pages/search/hooks/useTrekResults.ts [hunk 5]
```diff
@@ -77,8 +77,8 @@ export const useTrekResults = (
     hasNextPage,
     hasPreviousPage,
     isFetchingNextPage,
-  } = useInfiniteQuery<SearchResults, Error>(
-    [
+  } = useInfiniteQuery<SearchResults, Error>({
+    queryKey: [
       'trekResults',
       JSON.stringify(parsedFiltersState),
```
Comment (mabhub) [inline frontend/src/components/pages/search/hooks/useTrekResults.ts:83]:
```suggestion
      parsedFiltersState,
```

TanStack Query do it itself already: https://tanstack.com/query/latest/docs/framework/react/guides/query-keys

File: frontend/src/components/pages/home/useHome.ts [hunk 6]
```diff
@@ -22,11 +22,11 @@ export const useHome = (): UseHome => {
 
   const commonDictionaries = useQueryCommonDictionaries(language);
 
-  const { data = [] } = useQuery<ActivitySuggestion[] | [], Error>(
-    ['activitySuggestions', `Suggestion-${activitySuggestionIds.join('-')}`, language],
-    () => getActivitySuggestions(suggestions, language, commonDictionaries),
-    { enabled: suggestions.length > 0 && commonDictionaries !== undefined },
-  );
+  const { data = [] } = useQuery<ActivitySuggestion[] | [], Error>({
+    queryKey: ['activitySuggestions', `Suggestion-${activitySuggestionIds.join('-')}`, language],
```
Comment (dtrucs) [inline frontend/src/components/pages/home/useHome.ts:26]:
```suggestion
    queryKey: ['activitySuggestions', activitySuggestionIds, language],
```

File: frontend/src/pages/search.tsx [hunk 7]
```diff
@@ -1,60 +1,76 @@
...
+          commonDictionaries,
+        ),
+    });
 
-    await queryClient.prefetchInfiniteQuery(
-      [
+    await queryClient.prefetchInfiniteQuery({
+      queryKey: [
         'trekResults',
         JSON.stringify(parsedInitialFiltersState),
```
Comment (dtrucs) [inline frontend/src/pages/search.tsx:75]:
```suggestion
        parsedInitialFiltersState,
```

File: frontend/src/pages/search.tsx [hunk 8]
```diff
@@ -63,7 +79,7 @@ export const getServerSideProps: GetServerSideProps = async context => {
         JSON.stringify(dateFilter),
```
Comment (dtrucs) [inline frontend/src/pages/search.tsx:79]:
```suggestion
        dateFilter,
```

File: frontend/src/components/pages/search/hooks/useTrekResults.ts [hunk 9]
```diff
@@ -87,7 +87,7 @@ export const useTrekResults = (
       JSON.stringify(dateFilter),
```
Comment (dtrucs) [inline frontend/src/components/pages/search/hooks/useTrekResults.ts:87]:
```suggestion
      dateFilter,
```

**Pull request merged**